drivers/gpu/drm/panel/panel-tianma-499.c

```diff
@@ -0,0 +1,389 @@
+		return ret;
+	}
+
+	ctx->enabled = false;
+	return 0;
+}
+
+static const struct drm_display_mode tianma_499_mode = {
+	.clock = (720 + 52 + 20 + 84) * (1280 + 20 + 8 + 20) * 60 / 1000,
+	.hdisplay = 720,
+	.hsync_start = 720 + 52,
+	.hsync_end = 720 + 52 + 20,
+	.htotal = 720 + 52 + 20 + 84,
+	.vdisplay = 1280,
+	.vsync_start = 1280 + 20,
+	.vsync_end = 1280 + 20 + 8,
+	.vtotal = 1280 + 20 + 8 + 20,
+	.vrefresh = 60,
+	.width_mm = 62,
+	.height_mm = 110,
+};
+
+static int tianma_499_get_modes(struct drm_panel *panel)
+{
+	struct drm_display_mode *mode;
+
+	mode = drm_mode_duplicate(panel->drm, &tianma_499_mode);
+	if (!mode)
+		return -ENOMEM;
+
+	drm_mode_set_name(mode);
+
+	mode->type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED;
+	panel->connector->display_info.width_mm = mode->width_mm;
+	panel->connector->display_info.height_mm = mode->height_mm;
+	drm_mode_probed_add(panel->connector, mode);
+
+	return 1;
+}
+
+static const struct drm_panel_funcs tianma_499_panel_funcs = {
+	.disable = tianma_499_disable,
+	.unprepare = tianma_499_unprepare,
+	.prepare = tianma_499_prepare,
+	.enable = tianma_499_enable,
+	.get_modes = tianma_499_get_modes,
+};
+
+static int tianma_499_send_brightness(struct mipi_dsi_device *dsi, u16 brightness)
+{
+	ssize_t err;
+	u8 data[] = { MIPI_DCS_SET_DISPLAY_BRIGHTNESS,
+		      (brightness & 0xf00) >> 8, brightness & 0xff };
+
+	dsi->mode_flags &= ~MIPI_DSI_MODE_LPM;
+
+	err = mipi_dsi_dcs_write_buffer(dsi, data, sizeof(data));
+	if (err < 0)
+		return err;
+
+	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
+
+	return 0;
+}
+
+static int tianma_499_bl_update_status(struct backlight_device *bl)
+{
+	struct mipi_dsi_device *dsi = bl_get_data(bl);
+	struct tianma_499 *ctx = mipi_dsi_get_drvdata(dsi);
+	u16 brightness = bl->props.brightness;
+
+	if (bl->props.power != FB_BLANK_UNBLANK ||
+	    bl->props.fb_blank != FB_BLANK_UNBLANK ||
+	    bl->props.state & (BL_CORE_SUSPENDED | BL_CORE_FBBLANK))
+		brightness = 0;
+
+	gpiod_set_value_cansleep(ctx->backlight_gpio, !!brightness);
+
+	return tianma_499_send_brightness(dsi, brightness);
+}
+
+static const struct backlight_ops tianma_499_bl_ops = {
+	.update_status = tianma_499_bl_update_status,
+};
+
+static struct backlight_device *
+tianma_499_create_backlight(struct mipi_dsi_device *dsi)
+{
+	struct device *dev = &dsi->dev;
+	struct backlight_properties props = {
+		.type = BACKLIGHT_RAW,
+		.brightness = 4095,
+		.max_brightness = 4095,
+	};
+
+	return devm_backlight_device_register(dev, dev_name(dev), dev, dsi,
+					      &tianma_499_bl_ops, &props);
+}
+
+static int tianma_499_probe(struct mipi_dsi_device *dsi)
+{
+	struct device *dev = &dsi->dev;
+	struct tianma_499 *ctx;
+	int ret;
+
+	ctx = devm_kzalloc(dev, sizeof(*ctx), GFP_KERNEL);
+	if (!ctx)
+		return -ENOMEM;
+
+	ctx->supplies[0].supply = "vsp";
+	ctx->supplies[1].supply = "vsn";
+	ret = devm_regulator_bulk_get(dev, ARRAY_SIZE(ctx->supplies),
+				     ctx->supplies);
```

Comment:
somehow a space got lost here for proper alignment

```
CHECK: Alignment should match open parenthesis
#350: FILE: drivers/gpu/drm/panel/panel-tianma-499.c:303:
+	ret = devm_regulator_bulk_get(dev, ARRAY_SIZE(ctx->supplies),
+				     ctx->supplies);
```

```suggestion
				      ctx->supplies);
```